flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala

```diff
@@ -1288,7 +1295,8 @@ object ScalarOperators {
       nullCheck: Boolean,
       resultType: TypeInformation[_],
       left: GeneratedExpression,
-      right: GeneratedExpression)
+      right: GeneratedExpression,
+      compareNull: Boolean)
```

Comment:
@xueyumusic After looking at the amount of changes, I would suggest to leave `generateOperatorIfNotNull` untouched. This method is correct as it is. If you need some special null comparing logic. We can create a specialized method only for `equals`. We could create a method `generateDistinctFrom` that wraps `generateEquals`. What do you think?

Reply:
Yes, I think so, `generateOperatorIfNotNull` is a low level common method. I updated the code, thanks~ @twalthr